py2many/exceptions.py

```diff
@@ -1,7 +1,7 @@
 import ast
 
 
-class AstErrorBase:
+class AstErrorBase(Exception):
```

Comment:
this was intentionally not done so that subclasses need to explicitly choose which exception type they derive from, otherwise they are unthrowable.

Reply:
What does super() mean on line 8 without a base class?

Reply:
All classes have `object` as their base.  In Python 3 that became implicit, but in Python 2 if there was no explicit `object` the class was a `type` instead of a `class`.
https://stackoverflow.com/questions/35958961/class-vs-type-in-python

Reply:
>         super().__init__(msg)

The constructor of object doesn't take `msg` though, which is what the type checker was complaining about. 

We either remove this line, or have a base class that takes a message.

Reply:
ah, now I get you.  Well this code is correct, and and extra base doesnt make sense just to force subclasses to choice a more specific Exception subclass.  There are other ways to force that.

Reply:
Plugins have the same issue. They assume that `self` is subclassed from `$LANGTranspiler`, but type checker complains because it doesn't see that.

Swift has the concept of `Extensions` to describe this pattern. We have to look into what python type checkers are willing to do to support something similar.